IdentityCore/src/telemetry/request_telemetry/MSIDCurrentRequestTelemetry.m

```diff
@@ -29,17 +29,70 @@ @implementation MSIDCurrentRequestTelemetry
 
 - (NSString *)telemetryString
 {
-    return @"";
+    return [self serializeCurrentTelemetryString];
 }
 
-- (instancetype)initWithTelemetryString:(__unused NSString *)csvString error:(__unused NSError **)error
+- (instancetype)initWithTelemetryString:(NSString *)telemetryString error:(NSError **)error
 {
     self = [super init];
     if (self)
     {
+        NSError *internalError;
+        [self deserializeCurrentTelemetryString:telemetryString error:&internalError];
         
+        if (internalError) {
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"Failed to initialize server telemetry, error: %@", MSID_PII_LOG_MASKABLE(internalError));
+            if (error) *error = internalError;
+            return nil;
+        }
     }
     return self;
 }
 
+#pragma mark - Private
+
+-(NSString *)serializeCurrentTelemetryString
+{
+    int forceRefreshValue = (self.forceRefresh ? 1 : 0);
+    NSString *telemetryString = [NSString stringWithFormat:@"%ld|%ld,%d|", self.schemaVersion, self.apiId, forceRefreshValue];
```

Comment:
Just as an idea, instead of hardcoding string format, consider using built-in string formatting methods (e.g. componentsJoinedByString:https://developer.apple.com/documentation/foundation/nsarray/1412075-componentsjoinedbystring?language=objc). Number of components is pretty small right now, but will grow over time and maintaining a hardcoded string format might become a little bit tedious :)

For example, you could have a generic telemetry object with following properties:
- defaultFields (NSArray<NSString *> *)
- platformFields (NSArray<NSString *> *)

Then to serialize, all you need to do is have [NSString stringWithFormat:@"%@|%@|%@", self.schemaVersion, [self.defaultFields componentsJoinedByString:@","], [self.platformFields componentsJoinedByString:@","]]

You could extend it to have each components in array implement their own description method. For example, you could store ErrorsInfo object in defaultFields that implements "description" method and would create a string based on its components. We could later add other objects that all know how to serialize themselves.

Reply:
This was a great idea, I created a separate object to keep serialization for each set of fields together. This way any changes to serialization schema for default fields, platform fields, or error info will be handled in the same place for either telemetry object.